addons/web/static/tests/helpers/mock_server.js

```diff
@@ -528,117 +528,197 @@ var MockServer = Class.extend({
         return modelFields;
     },
     /**
-     * Simulate a call to the 'search_panel_select_range' method.
+     * Simulate a call to the search_panel_local_counters method.
+     * A model is added to the params to perform the adequate read_group.
      *
-     * Note that the implementation assumes that 'parent_id' is the field that
-     * encodes the parent relationship.
+     * @param {string} model
+     * @param {string} fieldName
+     * @param {Object} kwargs
+     * @returns {Object}
+     */
+    _mockSearchPanelLocalCounters: function (model, fieldName, kwargs) {
+        const countDomain = Domain.prototype.normalizeArray([
+            ...(kwargs.search_domain || []),
+            ...(kwargs.category_domain || []),
+            ...(kwargs.filter_domain || []),
+            ...(kwargs.group_domain  || []),
+            [fieldName, '!=', false],
+        ]);
+        const field = this.data[model].fields[fieldName];
+        let getGroupId;
+        if (field.type === 'many2one') {
+            getGroupId = group => group[fieldName][0];
+        } else {
+            getGroupId = group => group[fieldName];
+        }
+        const groups = this._mockReadGroup(model, {
+            domain: countDomain,
+            fields: [fieldName],
+            groupby: [fieldName],
+        });
+        const localCounters = {};
+        for (const group of groups) {
+            const groupId = getGroupId(group);
+            localCounters[groupId] = group[fieldName + '_count'];
+        }
+        return localCounters;
+    },
+    /**
+     * Simulate a call to the search_panel_global_counters method.
+     *
+     * @param {Map} valuesRange
+     * @param {string} parentName
+     */
+    _mockSearchPanelGlobalCounters: function (valuesRange, parentName) {
+        const localCounters = [...valuesRange.keys()].map(id => valuesRange.get(id).count);
+        for (let [id, values] of valuesRange.entries()) {
+            const count = localCounters[id];
```

Comment:
equivalent to `const count = values.count;`, right?

Reply:
nope (see message in related python code)